### PR TITLE
chore: remove version numbers from generated PB Go file

### PR DESCRIPTION
**Pull request by fiam**
Although version numbers could provide some value to detect version
incompatibility, we're only using PB to generate schema definitions
for JSON files, so this is a minor concern. Additionally, we do not
have version numbers in the TS side (because the generator doesn't
write them down). On the other hand, including the version number
causes significant friction for contributors using slightly different
PB versions while checking that the generated files are up to date)

### CODE REVIEW

File: types/generate.sh [hunk 1]
```diff
@@ -1,9 +1,19 @@
...
   --go_opt=module=github.com/wundergraph/wundergraph \
   --go_opt=paths=import \
   --experimental_allow_proto3_optional \
   wundernode_config.proto
+
+outfile='../pkg/wgpb/wundernode_config.pb.go'
+
+# Remove version numbers. Generating the backup is required
+# for compatibility with both GNU and BSD sed variants.
+sed -i.bak '2,4d' ${outfile}
```
Comment (StarpTech) [inline types/generate.sh:18]: This thread convinced me. https://github.com/golang/protobuf/issues/1185

Your regex is a bit dangerous. Could you use this one:

```sh
sed -i -e '1h;2,$H;$!d;g' -re 's|// versions.+// source:|// source:|' pkg/wgpb/*.pb.go
```

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: types/generate.sh (hunk 1 above)
Comment (StarpTech) [reply types/generate.sh:18]: Whereby this is still valid https://github.com/golang/protobuf/issues/1185#issuecomment-1305074470

> Note that this does not apply semver; the promise is expressed in terms of a time window.

File: types/generate.sh [hunk 2]
```diff
@@ -1,9 +1,18 @@
...
   --go_out=.. \
   --go_opt=module=github.com/wundergraph/wundergraph \
   --go_opt=paths=import \
   --experimental_allow_proto3_optional \
   wundernode_config.proto
+
+outfile="../pkg/wgpb/wundernode_config.pb.go"
+tmp="${outfile}.tmp"
+
+# Remove comments with version numbers
```
Comment (StarpTech) [inline types/generate.sh:17]: Could you add comment why we are doing this?

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: types/generate.sh (hunk 1 above)
Comment (fiam) [reply types/generate.sh]: Changed to grep

File: types/generate.sh (hunk 2 above)
Comment (fiam) [reply types/generate.sh:17]: I've added a comment explaining the rationale for dropping the version numbers as well as for choosing grep as the tool to do it. PTAL

**Review by StarpTech — APPROVED** [review]
LGTM

**Review by jensneuse — APPROVED** [review]

**Pull request merged**